airflow/providers/google/cloud/hooks/cloud_run.py

```diff
@@ -0,0 +1,233 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a Google Cloud Run Hook."""
+from __future__ import annotations
+
+import json
+import time
+from typing import Any, Callable, Dict, List, Sequence, Union, cast
+
+from google.api_core.client_options import ClientOptions
+from googleapiclient.discovery import build
+
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+DEFAULT_CLOUD_RUN_REGION = "us-central1"
+
+
+class CloudRunJobSteps:
+    """
+    Helper class with Cloud Run job status.
+    Reference: https://cloud.google.com/run/docs/reference/rest/v1/namespaces.jobs#JobStatus
+    """
+
+    EXEC_STEP_COMPLETED = "Completed"
+    EXEC_STEP_RESOURCES_AVAILABLE = "ResourcesAvailable"
+    EXEC_STEP_STARTED = "Started"
+    AWAITING_STEPS = {EXEC_STEP_STARTED, EXEC_STEP_RESOURCES_AVAILABLE}
+    ALL_STEPS = [EXEC_STEP_COMPLETED, EXEC_STEP_RESOURCES_AVAILABLE, EXEC_STEP_STARTED]
+
+
+class _CloudRunJobExecutionController(LoggingMixin):
+    """
+    Interface for communication with Google API.
+
+    :param cloud_run: Discovery resource
+    :param project_id: The Google Cloud Project ID.
+    :param execution_id: ID of a Cloud Run Job execution.
+    :param poll_sleep: The status refresh rate for pending operations.
+    :param num_retries: Maximum number of retries in case of connection problems.
+    :param wait_until_finished: If True, wait for the end of pipeline execution before exiting.
+        If False,it only submits job and check once is job not in terminal state.
```

Comment:
```suggestion
        If False, it only submits job and check once is job not in terminal state.
```

Docs spellcheck complain about missing whitespace after comma